client/src/home/index.js

```diff
@@ -68,29 +78,52 @@ export default function HomePage() {
   };
 
   const handlePathInputSubmit = async () => {
-    getStationsNearPath(startAddress, endAddress, maxDistance).then(
-      (response) => {
-        const stationData = response.stations.map((station) => {
-          return {
-            sid: station.sid,
-            location: {
-              y: station.latitude,
-              x: station.longitude,
-            },
-          };
-        });
-        // console.log(response);
-        // console.log(stationData);
-        setStations(stationData);
+    setCurLocation({});
+    setPath([]);
 
-        if (!response || response.length === 0) {
-          setStationFormError(true);
-          setStationFormErrorText(
-            "No stations found given the specified distance and locations. Please try again with different distance or locations."
-          );
-        }
+    Promise.all([
+      getCoordinatesFromAddress(startAddress),
+      getCoordinatesFromAddress(endAddress),
+    ])
+      .then(([srcCoor, destCoor]) => {
+        setSrcLocation(srcCoor);
+        setDestLocation(destCoor);
+      })
+      .catch(() => {
+        setPathFormError(true);
+        setPathFormErrorText(
+          "Invalid address. Please try again with a valid address."
```

Comment:
This error message seems too vague. What are the possible reasons for errors here? It may be because the address is not complete enough. Try giving more specific instructions, such as "please include the city name, zip code", etc.